lib/instana/instrumentation/sidekiq-client.rb

```diff
@@ -14,8 +14,17 @@ def call(worker_class, msg, queue, _redis_pool)
         # Temporary until we move connection collection to redis
         # instrumentation
         Sidekiq.redis_pool.with do |client|
-          opts = client.respond_to?(:connection) ? client.connection : client.client.options
-          kv_payload[:'sidekiq-client'][:'redis-url'] = "#{opts[:host]}:#{opts[:port]}"
+          sidekiq_version = Gem::Specification.find_by_name('sidekiq').version
+          host, port = if sidekiq_version >= Gem::Version.new('7.0') && client.respond_to?(:config) && client.config.respond_to?(:host) && client.config.respond_to?(:port)
+                         [client.config.host, client.config.port]
+                       elsif client.respond_to?(:connection)
+                         [client.connection[:host], client.connection[:port]]
+                       elsif client.respond_to?(:client) && client.client.respond_to?(:options)
+                         [client.client.options[:host], client.client.options[:port]]
+                       else
+                         ['Unknown', 'Unknown']
```

Comment:
Won't this break something? Aren't there default values, like `localhost` and a specific port number to use here?